ibis/backends/tests/test_client.py

```diff
@@ -295,3 +295,27 @@ def test_insert_overwrite_from_expr(
     con.insert(temp_table, obj=from_table, overwrite=True)
     assert len(temporary.execute()) == 3
     tm.assert_frame_equal(temporary.execute(), from_table.execute())
+
+
+@pytest.mark.only_on_backends(
+    SQLALCHEMY_BACKENDS, reason="run only if backend is SQLAlchemy based",
+)
+def test_list_databases(con):
+    # Every backend has its own databases
+    TEST_DATABASES = {
+        'SQLiteClient': ['main', 'base'],
+        'PostgreSQLClient': ['postgres', 'ibis_testing'],
+        'MySQLClient': ['ibis_testing', 'information_schema'],
+    }
+    assert con.list_databases() == TEST_DATABASES[con.__class__.__name__]
+
+
+@pytest.mark.only_on_backends(
+    set(SQLALCHEMY_BACKENDS) - {'postgres'},
+    reason="run only if backend is SQLAlchemy based, except postgres which "
+    "which has schemas different than databases",
+)
+def test_list_schemas(con):
+    with pytest.warns(FutureWarning):
+        schemas = con.list_schemas()
```

Comment:
why is this the case?

Reply:
Schemas as a PostgreSQL thing, and not common in many other database servers. They don't exist in SQLite or MySQL. And `list_schemas` is not present in SQL backends, only in SQLAlchemy backends.

What I want is to not have `list_schemas`  as an Ibis thing (it is now partly, as a SQLAlchemy method), and instead, leave it only as something the PostgreSQL backend implements. The warning is to not delete it directly, but to warn on Ibis 2.0, and remove it later. Does it sound reasonable?